From efa135aa633a008a4a9e1df936837e4e4e0262f6 Mon Sep 17 00:00:00 2001
From: Albert Chen <R65187@freescale.com>
Date: Mon, 13 Oct 2008 11:34:27 +0800
Subject: [PATCH] ENGR00094850 : poor USB performance if apply peripheral patch

Mx37: poor USB performance if if apply peripheral patch.
  just add the iram size, could make much better than before.

Signed-off-by: Albert Chen
---
 drivers/usb/gadget/arcotg_udc.h |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/gadget/arcotg_udc.h b/drivers/usb/gadget/arcotg_udc.h
index cc7580f..fc75890 100644
--- a/drivers/usb/gadget/arcotg_udc.h
+++ b/drivers/usb/gadget/arcotg_udc.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2004-2008 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2004-2008 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -29,7 +29,8 @@
 
 /* Iram patch */
 #ifdef CONFIG_USB_STATIC_IRAM_PPH
-#define IRAM_TD_PPH_SIZE	1024	/* size of 1 qTD's buffer */
+/* size of 1 qTD's buffer,one is for BULK IN and other is BULK OUT */
+#define IRAM_TD_PPH_SIZE	(USB_IRAM_SIZE / 2)
 #define IRAM_PPH_NTD	2	/* number of TDs in IRAM  */
 #else
 #define IRAM_TD_PPH_SIZE	0
-- 
1.5.4.4

